src/server.c

```diff
@@ -474,9 +486,12 @@ dictType dbDictType = {
     NULL,                 /* key dup */
     NULL,                 /* val dup */
     dictSdsKeyCompare,    /* key compare */
-    dictSdsDestructor,    /* key destructor */
+    NULL,                 /* key is embedded in the dictEntry and freed internally */
     dictObjectDestructor, /* val destructor */
     dictResizeAllowed,    /* allow to resize */
+    .keyLen = dictSdsKeyLen,
+    .keyToBytes = dictSdsKeyToBytes,
```

Comment:
```suggestion
    .keyCopy = dictSdsKeyCopy,
```

Reply:
In the context of a dictionary, I don't think `keyCopy` makes sense as an API name here. I would assume keyCopy creates a copy of the Key given something like a dictionary entry. I think the same is also true about the keyLen API. I would almost prefer to add embedded, something like: 
```suggestion
    .embededKeyLen = whatever,
    .embedKey = whatever,
```

Reply:
Yeah `embedKey` is better but maybe we can remove `embedKeyLen`? I am thinking that we can have 
`dt->embedKey(NULL, 0, key, &entry->key_header_size)` return both the header size and the required buffer size, similar to `writePingExt`, https://github.com/valkey-io/valkey/blob/unstable/src/cluster_legacy.c#L3680

Reply:
I like that idea, having one API seems like a good approach.